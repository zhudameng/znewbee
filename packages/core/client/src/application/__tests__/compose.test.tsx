import React from 'react';
import { render } from '@testing-library/react';
import { compose } from '../';

describe('compose', () => {
  it('case 1', () => {
    const A: React.FC = (props) => (
      <div>
        <h1>A</h1>
        {props.children}
      </div>
    );
    const App = compose(A)();
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('case 2', () => {
    const A: React.FC = (props) => (
      <div>
        <h1>A</h1>
        {props.children}
      </div>
    );
    const B: React.FC = (props) => (
      <div>
        <h1>B</h1>
        {props.children}
      </div>
    );
    const App = compose(A)(B);
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('case 3', () => {
    const A: React.FC = (props) => (
      <div>
        <h1>A</h1>
        {props.children}
      </div>
    );
    const App = compose([A])();
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('case 4', () => {
    const A: React.FC = (props) => (
      <div>
        <h1>A</h1>
        {props.children}
      </div>
    );
    const B: React.FC = (props) => (
      <div>
        <h1>B</h1>
        {props.children}
      </div>
    );
    const C: React.FC = (props) => (
      <div>
        <h1>C</h1>
        {props.children}
      </div>
    );
    const App = compose(A, B)(C);
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('case 5', () => {
    const A: React.FC<any> = (props) => (
      <div>
        <h1>A {props.name}</h1>
        {props.children}
      </div>
    );
    const B: React.FC<any> = (props) => (
      <div>
        <h1>B {props.name}</h1>
        {props.children}
      </div>
    );
    const C: React.FC<any> = (props) => (
      <div>
        <h1>C</h1>
        {props.children}
      </div>
    );
    const App = compose(A, [B, { name: '1' }])(C);
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
