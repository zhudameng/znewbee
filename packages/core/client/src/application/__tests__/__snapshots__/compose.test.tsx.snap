// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compose case 1 1`] = `
<div>
  <div>
    <h1>
      A
    </h1>
  </div>
</div>
`;

exports[`compose case 2 1`] = `
<div>
  <div>
    <h1>
      A
    </h1>
    <div>
      <h1>
        B
      </h1>
    </div>
  </div>
</div>
`;

exports[`compose case 3 1`] = `
<div>
  <div>
    <h1>
      A
    </h1>
  </div>
</div>
`;

exports[`compose case 4 1`] = `
<div>
  <div>
    <h1>
      A
    </h1>
    <div>
      <h1>
        B
      </h1>
      <div>
        <h1>
          C
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`compose case 5 1`] = `
<div>
  <div>
    <h1>
      A 
    </h1>
    <div>
      <h1>
        B 
        1
      </h1>
      <div>
        <h1>
          C
        </h1>
      </div>
    </div>
  </div>
</div>
`;
